compiler/crates/relay-bin/src/main.rs

```diff
@@ -36,6 +41,10 @@ struct Opt {
     #[clap(long, short)]
     watch: bool,
 
+    /// Run the LSP server
+    #[clap(long, short)]
+    lsp: bool,
```

Comment:
I think that rather than a flag, we should use an optional subcommand. That would allow us to have flags which are LSP specific. Will also allow for future compiler subcommands (I could imagine one that would dump stats about your codebase or something).

https://docs.rs/structopt/latest/structopt/#subcommands

I don't think StructOpt supports default subcommands, so we'll still need to handle the fact that some global options (intended for the compile) don't make sense in the context of the LSP, but I think that's okay.